aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+                if dim in scene_dims_dict
+            ]
+            retrieved_chunk = np.stack(planes).reshape(
+                np_array_for_indices.shape + tuple(new_chunk_shape)
+            )
+
+            # Remove extra dimensions if they were not requested
+            remove_dim_ops_list: List[Union[int, slice]] = []
+            for index in retrieve_indices:
+                if isinstance(index, int):
+                    remove_dim_ops_list.append(0)
+                else:
+                    remove_dim_ops_list.append(slice(None, None, None))
+
+            # Remove extra dimensions by using dim ops
+            retrieved_chunk = retrieved_chunk[tuple(remove_dim_ops_list)]
+
+            return retrieved_chunk
+
+    def _create_dask_array(
+        self, czi: CziFile, selected_scene_dims: List[str]
+    ) -> xr.DataArray:
+        """
+        Creates a delayed dask array for the file.
+
+        Parameters
+        ----------
+        czi: CziFile
+            An open CziFile for processing.
+        selected_scene_dims: List[str]
+            The dimensions for the scene to create the dask array for
+
+        Returns
+        -------
+        image_data: da.Array
+            The fully constructed and fully delayed image as a Dask Array object.
+        """
+        # Always add the plane dimensions if not present already
+        for dim in REQUIRED_CZI_CHUNK_DIMS:
+            if dim not in self.chunk_by_dims:
+                self.chunk_by_dims.append(dim)
+
+        # Safety measure / "feature"
+        self.chunk_by_dims = [d.upper() for d in self.chunk_by_dims]
+
+        # Construct the delayed dask array
+        dims_shape = CziReader._dims_shape_to_scene_dims_shape(
+            czi.get_dims_shape(),
+            scene_index=self.current_scene_index,
+            consistent=czi.shape_is_consistent,
+        )
+
+        valid_dims = []
+        selected_scene_shape: List[int] = []
+        for dim in selected_scene_dims:
+            if dim in dims_shape.keys():
+                valid_dims.append(dim)
+                if dim == DimensionNames.MosaicTile:
+                    selected_scene_shape.append(
+                        dims_shape[DimensionNames.MosaicTile][1]
+                    )
+                elif dim == DimensionNames.Time:
+                    selected_scene_shape.append(dims_shape[DimensionNames.Time][1])
+                elif dim == DimensionNames.Channel:
+                    selected_scene_shape.append(dims_shape[DimensionNames.Channel][1])
+                elif dim == DimensionNames.SpatialZ:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialZ][1])
+                elif dim == DimensionNames.SpatialY:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialY][1])
+                elif dim == DimensionNames.SpatialX:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialX][1])
```

Comment:
dims_shape = { "T": (start, length) ... } so we want the length each time.
Cleaner way to do this? The start is a little pedantic, it's almost always zero but the spec says it can be non-zero.

Reply:
Ohhhhh. I just wish it was a NamedTuple or some object instead of 0 and 1. Named attributes are good attributes.

Reply:
If you don't want to use a `NamedTuple` I think just adding a comment that says "item at index 1 is the length of the dim" would be fine as well. I just didn't know what it was.